backend/src/main/java/gov/cdc/usds/simplereport/config/AzureTestEventReportingQueueConfiguration.java

```diff
@@ -8,38 +12,78 @@
 import gov.cdc.usds.simplereport.api.model.errors.TestEventSerializationFailureException;
 import gov.cdc.usds.simplereport.db.model.TestEvent;
 import gov.cdc.usds.simplereport.properties.AzureStorageQueueReportingProperties;
+import gov.cdc.usds.simplereport.service.AzureStorageQueueFhirReportingService;
 import gov.cdc.usds.simplereport.service.AzureStorageQueueTestEventReportingService;
 import gov.cdc.usds.simplereport.service.TestEventReportingService;
-import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Profile;
 
 @Slf4j
 @Configuration
 class AzureTestEventReportingQueueConfiguration {
-  @Bean
-  @Primary
+  @Bean("csvQueueReportingService")
   @ConditionalOnProperty(
       value = "simple-report.azure-reporting-queue.enabled",
       havingValue = "true")
   TestEventReportingService storageQueueReportingService(
-      ObjectMapper mapper, QueueAsyncClient queueClient) {
+      ObjectMapper mapper, @Qualifier("csvQueue") QueueAsyncClient queueClient) {
     log.info("Configured for queue={}", queueClient.getQueueName());
     return new AzureStorageQueueTestEventReportingService(mapper, queueClient);
   }
 
+  @Bean("fhirQueueReportingService")
+  @ConditionalOnProperty(
+      value = "simple-report.azure-reporting-queue.fhir-queue-enabled",
+      havingValue = "true")
+  TestEventReportingService fhirQueueReportingService(
+      FhirContext context, @Qualifier("fhirQueueClient") QueueAsyncClient queueClient) {
+    log.info("Configured for queue={}", queueClient.getQueueName());
+    return new AzureStorageQueueFhirReportingService(context, queueClient);
+  }
+
   @Bean
-  @ConditionalOnMissingBean
-  TestEventReportingService noOpReportingService() {
+  FhirContext fhirContext() {
+    return FhirContext.forR4();
+  }
+
+  @Profile(PROD)
+  @Primary
+  @Bean(name = "csvQueueReportingService")
+  @ConditionalOnMissingBean(name = "csvQueueReportingService")
+  TestEventReportingService noOpCSVReportingService() {
     return new NoOpReportingService();
   }
 
-  @Bean
+  @Profile(PROD)
+  @Bean(name = "fhirQueueReportingService")
+  @ConditionalOnMissingBean(name = "fhirQueueReportingService")
+  TestEventReportingService noOpFhirReportingService() {
+    return new NoOpReportingService();
+  }
+
+  @Profile(NOT + PROD)
```

Comment:
You can also just add the "!" here directly if you prefer - we're doing that in a couple of other services already. I don't have a strong preference; just letting you know.